pkgs/build-support/docker/default.nix

```diff
@@ -505,6 +505,12 @@ rec {
         tar -rpf temp/layer.tar --hard-dereference --sort=name --mtime="@$SOURCE_DATE_EPOCH" \
           --owner=0 --group=0 --no-recursion --files-from newFiles
 
+        # Append nix/store directory to the layer so that when the layer is loaded in the
+        # image /nix/store has read permissions for non-root users.
+        mkdir -p nix/store
+        chmod -R 555 nix
+        tar -rpf temp/layer.tar --mtime="@$SOURCE_DATE_EPOCH" --owner=0 --group=0 ./nix
```

Comment:
This introduces code about the Nix store into `buildImage`, I think it's been agnostic about the content until this. I don't think everything which uses `buildImage` will necessarily use `/nix`. I'd prefer to keep references outside of this function.

Reply:
Yeah I like that better. Not great, but I think it's better.

Reply:
I think you can test if `layerClosure` is non empty to create `/nix` and `/nix/store`. Moreover, we should also test if `/nix` and `/nix/store` are not in `baseFiles` (not already existing in parent layers).